pandas/core/reshape/merge.py

```
@@ -1954,42 +1954,65 @@ def _validate_left_right_on(self, left_on, right_on):
+            return list(dups.unique()[:limit])
 
         # Check data integrity
         if validate in ["one_to_one", "1:1"]:
             if not left_unique and not right_unique:
+                combined_keys = list(left_keys.append(right_keys))
+                sample = sample_duplicates(combined_keys, limit=10)
                 raise MergeError(
                     "Merge keys are not unique in either left "
                     "or right dataset; not a one-to-one merge"
+                    f"Offending keys (sample): {sample}"
                 )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)